inc/common/purge.php

```diff
@@ -479,10 +481,20 @@ function do_admin_post_rocket_purge_cache() { // phpcs:ignore WordPress.NamingCo
 							'sslverify'  => apply_filters( 'https_local_ssl_verify', false ), // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals
 						]
 					);
-					wp_safe_remote_get(
-						home_url( $lang ),
-						$args
-					);
+
+					wp_safe_remote_get( $home_url, $args );
+
+					$preload_process = new WP_Rocket\Preload\Full_Process();
+
+					if ( $preload_process->is_mobile_preload_enabled() ) {
+						if ( empty( $args['user-agent'] ) ) {
+							$args['user-agent'] = $preload_process->get_item_user_agent( [ 'mobile' => true ] );
+						} else {
+							$args['user-agent'] .= ' iPhone';
+						}
+
+						wp_safe_remote_get( $home_url, $args );
+					}
```

Comment:
Alternative proposal: add a new `do_action()` here, and in the Homepage subscriber, you add a new method callback to do the mobile homepage preload if necessary.

I think it would make the separation of concerns better, and also be more performant (no need to re-instantiate the preload class)